### PR TITLE
[6.3] Added stubs around File Repository feature

### LINKED ISSUE #4970: Create stubs for File Repository features on UI

### PR DESCRIPTION
**Pull request by renzon**
close #4970

### CODE REVIEW

File: tests/foreman/ui/test_repository.py [hunk 1]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+
+        :id: 9f90c42e-a077-4f0a-b5ec-2b342ff2d9fe
+
+        :Setup:
+            1. Navigate do Content -> Products
+            2. Create a Product
+
+        :Steps:
+            1. Click "New Repository" button
+            2. Define a valid name and chose "type" file
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: typo: Define a valid name and choose "type" file

File: tests/foreman/ui/test_repository.py [hunk 2]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_name(self):
+        """Check File Repository creation
+
+        :id: 9f90c42e-a077-4f0a-b5ec-2b342ff2d9fe
+
+        :Setup:
+            1. Navigate do Content -> Products
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: typo: Navigate to Content -> Products

File: tests/foreman/ui/test_repository.py [hunk 3]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+
+    @stubbed
+    @tier1
+    def test_positive_upload_file(self):
+        """Check arbitrary file can be uploaded to File Repository
+
+        :id: 2a9b74a4-cba8-430d-a699-fea543496466
+
+        :Setup:
+            1. Navigate do Content -> Products
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: typo: Navigate to Content -> Products

File: tests/foreman/ui/test_repository.py [hunk 4]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+            2. Create a Product
+            3. Create a File Repository
+            4. Click on created repository
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: typo: Click on the created repository

File: tests/foreman/ui/test_repository.py [hunk 5]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+
+        :Steps:
+            1. Click "Browse" button and select a file
+            2. Click Upload
+            3. Click the link in "Files" on "Content Type" table
+            4. Check uploaded file is listed
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: we can omit this step here, as it's the same as expectedresults

File: tests/foreman/ui/test_repository.py [hunk 6]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+
+        :expectedresults: uploaded file is available under File Repository
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: The uploaded file is available under File Repository

File: tests/foreman/ui/test_repository.py [hunk 7]
```diff
@@ -2035,3 +2035,297 @@ def test_positive_git_view_content(self):
...
+            2. Create a Product
+            3. Create a File Repository
+            4. Click on created repository
+
+        :Steps:
+            1. Click "Browse" button and select a large file
+                (3GB - can use a Sallite ISO)
+            2. Click Upload
+            3. Click the link in "Files" on "Content Type" table
+            4. Check uploaded file is listed
```
Comment (ldjebran) [inline tests/foreman/ui/test_repository.py]: we can omit this step here, as it's the same as expectedresults

File: tests/foreman/ui/test_repository.py [hunk 8]
```diff
@@ -2035,3 +2035,286 @@ def test_positive_git_view_content(self):
...
+
+        :id: 9f90c42e-a077-4f0a-b5ec-2b342ff2d9fe
+
+        :Setup:
+            1. Navigate to Content -> Products
+            2. Create a Product
+
+        :Steps:
+            1. Click "New Repository" button
+            2. Define a valid name and choose "type" file
```
Comment (sghai) [inline tests/foreman/ui/test_repository.py]: Choose "file" type repo ?

File: tests/foreman/ui/test_repository.py [hunk 9]
```diff
@@ -2035,3 +2035,286 @@ def test_positive_git_view_content(self):
...
+    @tier1
+    def test_positive_upload_file(self):
+        """Check arbitrary file can be uploaded to File Repository
+
+        :id: 2a9b74a4-cba8-430d-a699-fea543496466
+
+        :Setup:
+            1. Navigate to Content -> Products
+            2. Create a Product
+            3. Create a File Repository
```
Comment (sghai) [inline tests/foreman/ui/test_repository.py]: `Create a file repository w/o specifying repo URL`

**Comment by sghai** [comment]
ACK, a few nitpicks, non blocking comments.

File: tests/foreman/ui/test_repository.py (hunk 9 above)
Comment (renzon) [inline tests/foreman/ui/test_repository.py]: I changed this but am not sure why GH ui is not showing it here on conversation tab.

**Comment by sghai** [comment]
ACK

**Pull request merged**